diff --git a/dist/client/components/Changelog.vue b/dist/client/components/Changelog.vue
index c1716c10b04816ef22179ed738a7e4809aeaba76..28a094837ff5b7e198139daadbb1393bc406ef1e 100644
--- a/dist/client/components/Changelog.vue
+++ b/dist/client/components/Changelog.vue
@@ -25,7 +25,11 @@ const { commits, useHmr } = useChangelog()
 // The order of commits, defaults to true (descending order)
 const isDescending = ref(true)
 const toggleViewMore = ref(false)
-const lastChangeDate = ref<Date>(commits.value[0]?.date_timestamp ? toDate(commits.value[0]?.date_timestamp) : new Date())
+const lastChangeDate = computed<Date>(() => {
+  // Find the latest regular commit (excluding typst fake commits)
+  const last = commits.value.find(c => !c.tag)
+  return last?.date_timestamp ? toDate(last?.date_timestamp) : new Date()
+})
 
 const locale = computed<Locale>(() => {
   if (!options.locales || typeof options.locales === 'undefined')
diff --git a/dist/client/composables/changelog.mjs b/dist/client/composables/changelog.mjs
index f63b201c453809ad5a5797360ce3f55a70ee6581..0d710d3bb31e2a157358dc503b32d3f7cf8ddf24 100644
--- a/dist/client/composables/changelog.mjs
+++ b/dist/client/composables/changelog.mjs
@@ -2,6 +2,20 @@ import changelog from "virtual:nolebase-git-changelog";
 import { useData } from "vitepress";
 import { computed, ref, toValue } from "vue";
 import { isStringArray } from "../utils.mjs";
+
+import TYPST_TAGS_FOR_NOLEBASE from "virtual:zhtyp-guide-typst-version"
+
+const TYPST_FAKE_COMMITS = TYPST_TAGS_FOR_NOLEBASE.map((
+  { date_timestamp, tag, release_tag_url },
+) => ({
+  date_timestamp,
+  authors: [],
+  tag,
+  tags: [tag],
+  release_tag_url,
+  release_tags_url: [release_tag_url],
+}));
+
 export function useChangelog() {
   const { page } = useData();
   const gitChangelog = ref(changelog);
@@ -10,10 +24,17 @@ export function useChangelog() {
   const _commits = computed(() => {
     const currentPath = toValue(page.value.filePath);
     const allCommits = gitChangelog.value.commits;
-    const commits2 = allCommits.filter((c) => c.paths.includes(currentPath)) || [];
-    return commits2.sort((a, b) => b.date_timestamp - a.date_timestamp).filter((commit, index) => {
-      if (commit.tag && (!commits2[index + 1] || commits2[index + 1]?.tag))
-        return false;
+    const commits2 = TYPST_FAKE_COMMITS.concat(allCommits.filter((c) => c.paths.includes(currentPath)) || []);
+    return commits2.sort((a, b) => b.date_timestamp - a.date_timestamp).filter((commit, index, arr) => {
+      // Drop irrelevant tags
+      if (commit.tag) {
+        // `prev` & `next` are in the sense of time, not the position in the array
+        const next = arr[index - 1];
+        const prev = arr[index + 1];
+        // Keep a tag if and only if it's adjacent to a regular commit.
+        return (next && !next.tag) || (prev && !prev.tag);
+      }
+      // Keep all regular commits
       return true;
     });
   });
